### PR TITLE
RTC-14634

### PR DESCRIPTION
**Pull request by olofkallander**
Automatic fail over to another candidate when current one becomes unresponsive.

also fix for keeping conference alive on ICE activity on the transports.

### CODE REVIEW

**Review by reddvl1980 — APPROVED** [review]

File: transport/ice/IceSession.cpp [hunk 1]
```diff
@@ -984,22 +1050,50 @@ void IceSession::stateCheck(const uint64_t now)
...
     }
 }
 
 void IceSession::freezePendingProbes(uint64_t now)
 {
     for (auto& candidatePair : _candidatePairs)
     {
-        if (!candidatePair->nominated && candidatePair->state < CandidatePair::State::Failed &&
-            candidatePair->localCandidate.transportType == ice::TransportType::TCP)
+        if (!candidatePair->nominated && candidatePair->state <= CandidatePair::State::Succeeded)
```
Comment (RicardoMDomingues) [inline transport/ice/IceSession.cpp:1078]: Why did you remove the `candidatePair->localCandidate.transportType == ice::TransportType::TCP` condition?

I think probes are like a keep-alive, if we don't probe them, NAT gateways can close them potentially by inactivity and then if we need to re-probe they might fail.
